src/jdk.incubator.vector/share/classes/jdk/incubator/vector/VectorMathLibrary.java

```diff
@@ -0,0 +1,311 @@
+    private static class SLEEF implements Library {
+        static {
+            VectorSupport.loadNativeLibrary("sleef");
+        }
+
+        private static String suffix(VectorShape vshape) {
+            return (vshape.vectorBitSize() > 128 ? "sve" : "advsimd");
+        }
+
+        private static String precisionLevel(Operator op) {
+            return (op == HYPOT ? "u05" : "u10");
+        }
+
+        @Override
+        public String symbolName(Operator op, VectorSpecies<?> vspecies) {
+            return String.format("%s%s%d_%s%s", op.operatorName(),
+                                 (vspecies.elementType() == float.class ? "f" : "d"),
+                                 vspecies.length(),
+                                 precisionLevel(op),
+                                 suffix(vspecies.vectorShape()));
+        }
+
+        @Override
+        public boolean isSupported(Operator op, VectorSpecies<?> vspecies) {
+            Class<?> etype = vspecies.elementType();
+            if (etype != float.class && etype != double.class) {
+                return false; // only FP element types are supported
+            }
+            int maxLaneCount = VectorSupport.getMaxLaneCount(vspecies.elementType());
+            if (vspecies.length() > maxLaneCount) {
+                return false; // lacking vector support
+            }
+            if (vspecies.vectorBitSize() < 128) {
+                return false; // 64-bit vectors are not supported
+            }
+            if (op == TANH) {
+                return false; // skip due to performance considerations
+            }
+            return true;
+        }
+    }
+
+    private static final int SIZE = VectorSupport.VECTOR_OP_MATHLIB_LAST - VectorSupport.VECTOR_OP_MATHLIB_FIRST + 1;
+
+    private record Entry<T> (String name, MemorySegment entry, T impl) {}
+
+    private static final @Stable Entry<?>[][][] LIBRARY_ENTRIES = new Entry<?>[SIZE][LaneType.SK_LIMIT][VectorShape.SK_LIMIT]; // OP x SHAPE x TYPE
+
+    @ForceInline
+    private static <T> Entry<T> lookup(Operator op, int opc, VectorSpecies<?> vspecies, IntFunction<T> implSupplier) {
+        int idx = opc - VectorSupport.VECTOR_OP_MATHLIB_FIRST;
+        int elem_idx = ((AbstractSpecies<?>)vspecies).laneType.switchKey;
+        int shape_idx = vspecies.vectorShape().switchKey;
+        @SuppressWarnings({"unchecked"})
+        Entry<T> entry = (Entry<T>)LIBRARY_ENTRIES[idx][elem_idx][shape_idx];
+        if (entry == null) {
+            entry = constructEntry(op, opc, vspecies, implSupplier);
+            LIBRARY_ENTRIES[idx][elem_idx][shape_idx] = entry;
+        }
+        return entry;
+    }
+
+    @DontInline
+    private static
+    <E,T>
+    Entry<T> constructEntry(Operator op, int opc, VectorSpecies<E> vspecies, IntFunction<T> implSupplier) {
+        if (LIBRARY.isSupported(op, vspecies)) {
+            String symbol = LIBRARY.symbolName(op, vspecies);
+            MemorySegment addr = LOOKUP.find(symbol)
```

Comment:
It is better to use `LOOKUP.findOrThrow()` because it does not require lambda creation.

Reply:
Thanks, changed as you suggested. I introduced a try-catch block instead.